CONTRIBUTING.md

```diff
@@ -1,11 +1,51 @@
-# How to Contribute
+# eksctl :heart:s your contributions
```

Comment:
I big thing I repeatedly see when people are contributing is not understanding how our `types.go` results in an auto-generated file getting updated. Folks often hit merge conflicts regarding this file and don't know how to resolve it (the answer is to merge master in, pick any of the conflicts for the auto-generated files and just re-generate). Is having a list of gotchas/commonly hit problems while contributing worth having?

example of the above happening: https://github.com/weaveworks/eksctl/pull/2886#issuecomment-737319991

Reply:
yep i will add that, i've been asked about it too 👍

Reply:
(the answer is never to "merge master" :trollface: rebase ftw!)

Reply:
Merge master 4life :trollface:

Reply:
stop resolving stuff i haven't done yet jake!

rude

Reply:
did I?! Sorry!